impeller/entity/shaders/blending/advanced_blend.frag

```diff
@@ -41,13 +41,12 @@ void main() {
                                  v_dst_texture_coords  // texture coordinates
                                  ));
   dst *= blend_info.dst_input_alpha;
-  f16vec4 src = IPHalfUnpremultiply(
+  f16vec4 src =
       blend_info.color_factor > 0.0hf
           ? blend_info.color
-          : Sample(texture_sampler_src,  // sampler
+          : IPHalfUnpremultiply(Sample(texture_sampler_src,  // sampler
                    v_src_texture_coords  // texture coordinates
-                   ));
-  src *= blend_info.src_input_alpha;
+                   )) * blend_info.src_input_alpha;
```

Comment:
When applying `blend_info.src_input_alpha` to an unpremultiplied color, only the `a` component should be adjusted, otherwise the color components will become darker.

Reply:
done